enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,170 @@
+---
+title: cli-in-cluster-management
+authors:
+  - "@sallyom"
+reviewers:
+  - TBD
+  - "@soltysh"
+approvers:
+  - TBD
+  - "@soltysh"
+creation-date: 2019-12-03
+last-updated: 2020-03-05
+status: provisional
+---
+
+# CLI In-Cluster Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is describing the mechanism for how CLI authors can deliver tools to OpenShift clusters.  An in-cluster feature
+is needed to manage various CLIs available for OpenShift and related services.  
+The goal is for a user to discover, install and upgrade tools easily and from a single
+location.  
+
+Each component is responsible for building its artifacts and registering information regarding supplied binaries
+to a central index location and/or image.  Currently, that location is [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/).
```

Comment:
Why do we have to support 2 delivery paths and not just the image mechanism?

Reply:
we don't, I updated